src/monster.cpp

```diff
@@ -924,6 +923,9 @@ void monster::melee_attack(Creature &target, bool, const matec_id&) {
                             target.disp_name(true).c_str());
             }
         }
+        if( !is_hallucination() ) {
```

Comment:
hrm, the enemy isn't real, but you actually dodged, it seems like it should trigger anyway. (not directly related to your PR, so I'm not asking for a fix)